t/mojo/promise.t

```diff
@@ -516,4 +516,30 @@ subtest 'Wait for stopped loop' => sub {
   is_deeply \@results, ['wait'], 'promise resolved';
 };
 
+subtest 'Resolve to self' => sub {
+  my $promise = Mojo::Promise->new;
+  my (@results, @errors);
+  $promise->resolve($promise)->then(sub { @results = @_ }, sub { @errors = @_ });
+  Mojo::IOLoop->one_tick;
+  is_deeply \@results, [], 'promise not resolved';
+  cmp_ok scalar(@errors), '==', 1, 'correct number of errors';
+  like $errors[0], qr/^Chaining cycle detected for promise \S/, 'correct error';
+};
+
+subtest 'Resolve to unsettled promise' => sub {
```

Comment:
Is it resolve "to" or resolve "with" unsettled promise?

Reply:
I do not know. Couldn't find either of the two in past tests. I believe I haven't used both verbs in my commit.

But I believe "resolve to" conveys the message in a more clear way in this case, because "Resolve with unsettled promise" could also mean "calling `resolve` on an unsettled promise" which is not what I meant.

* This promise resolves to 5
* This promise resolves with 5

Maybe both expressions are valid, but I can't say. I'm not that good with English prepositions.

Reply:
Since I saw you were already using the preposition "with" in similar -although non-ambiguous- situations inside `mojo/promise.t`, I decided to keep the same preposition, and addressed the ambiguity by renaming "Resove to unsettled promise" to "Settle with unsettled promise" and added the `$p->reject($p2)` case inside that subtest, too. Also I renamed "Resolve to self" to "Resolve with self", as that was already non-ambiguous.

To make sure the case "Reject with self" is covered, I renamed the subtest "Resolve to self" to "Settle with self" and added the "Reject with self" case in there too.

Reply:
shadowcat-mst below provided the answer to your question: "resolve "to" or resolve "with"" (it is "resolve with"). I have thus renamed the subtests, and included more cases in them (with "reject" instead of "resolve").

Anything else you need to accept this PR?